### PR TITLE
DAT-83 async improvements

### PR DESCRIPTION
**Pull request by rcrichton**

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -2,7 +2,8 @@
   "api": {
     "username": "root@openhim.org",
     "password": "password",
-    "apiURL": "https://localhost:8080"
+    "apiURL": "https://localhost:8080",
+    "trustSelfSigned": false
```
Comment (hnnesv) [inline config/config.json]: I can't see a reference in the code for this option, is it needed, or am I just missing the usage?

File: config/mediator.json [hunk 2]
```diff
@@ -47,7 +47,13 @@
     {
       "displayName": "Upstream task polling URL",
       "description": "The URL to use when request task updates (only used if aync is true)",
```
Comment (hnnesv) [inline config/mediator.json]: typo: async

File: config/mediator.json [hunk 3]
```diff
@@ -47,7 +47,13 @@
...
-      "param": "upsteamTaskURL",
+      "param": "upstreamTaskURL",
+      "type": "string"
+    },
+    {
+      "displayName": "Upstream task summaries URL",
+      "description": "The URL to use when requesting a task summary once a task has been marked as completed (only used if aync is true)",
```
Comment (hnnesv) [inline config/mediator.json]: typo: async

File: index.js [hunk 4]
```diff
@@ -95,25 +102,58 @@ function forwardResponse(statusCode, body, adxAdapterID) {
     body: { code: statusCode, message: body },
     json: true
   };
-  request.put(options, (err) => {
+  request.put(options, (err, res) => {
     if (err) {
-      console.log(err.stack);
+      return winston.error('Unable to forward response to reciever', err);
```
Comment (hnnesv) [inline index.js]: typo: receiver

File: index.js [hunk 5]
```diff
@@ -95,25 +102,58 @@ function forwardResponse(statusCode, body, adxAdapterID) {
...
+    }
+    winston.info('Response forwarded to receiver', res.statusCode);
```
Comment (hnnesv) [inline index.js]: typo: receiver

File: config/config.json (hunk 1 above)
Comment (rcrichton) [inline config/config.json]: The api property is passed directly to the mediator-utils which then makes use of this.

File: index.js (hunk 5 above)
Comment (rcrichton) [inline index.js]: This one is correct I think

**Pull request merged**